Added FAQ section about overriding service arguments (#5487)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/DependencyInjection/Index.rst

    Call to a member function methodName() on null

> 
> 
> How to override service arguments?

Reviewer: ```suggestion

..  dependency-injection-override-service-arguments:

How to override service arguments?
```

I know the other sections are still missing anchors, however this anchor will enable us to create a permalink to your new section.

Reviewer: Thanks a lot for putting in the efford to improve the docs here! I have just some small remarks regarding ReST syntax and documentation best practices.

----------------------------------

Some services in the TYPO3 core use service arguments, which can be overridden by third-party extensions.  

> For example, the :php:`$rateLimiterFactory` argument in the :php:`PasswordRecoveryController` of the 

Reviewer: ```suggestion
For example, the :php:`$rateLimiterFactory` argument in the :php-short:`\TYPO3\CMS\FrontendLogin\ControllerPasswordRecoveryController` of the 
```

Using the FQN in combination with php-short will only print `ControllerPasswordRecoveryController` but enable users to open the class on GitHub with two clicks

> :php:`felogin` extension  uses a service with the ID :php:`feloginPasswordRecovery.rateLimiterFactory`. 

Reviewer: ```suggestion
:composer:`typo3/cms-felogin` extension  uses a service with the ID :yaml:`feloginPasswordRecovery.rateLimiterFactory`. 
```

This service is defined in the :php:`Services.yaml` file of :php:`ext:felogin` and includes a service 
argument named :php:`$config`, which specifies the configuration for the Symfony Rate Limiter used in 
the class.

A third-party extension can override the :php:`feloginPasswordRecovery.rateLimiterFactory` service in 

> its own :php:`Services.yaml` file, as shown in the example below:

Reviewer: ```suggestion
its own :file:`Services.yaml` file, as shown in the example below:
```

> .. code-block:: php

Reviewer: ```suggestion
.. code-block:: yaml
```

> .. code-block:: php
> 

Reviewer: ```suggestion
..  code-block:: php
    :caption: packages/my-extension/Configuration/Services.yaml

```

    feloginPasswordRecovery.rateLimiterFactory:
      class: Symfony\Component\RateLimiter\RateLimiterFactory
      arguments:
        $config:
          id: 'felogin-password-recovery'
          policy: 'sliding_window'
          limit: 10
          interval: '30 minutes'
        $storage: '@storage.cachingFramework'

It is important that the 3rd party extension is loaded **after** the extension whose service it overrides.

> This can be achieved by requiring the original extension as a dependency in the :php:`composer.json` file

Reviewer 2: Do we not have a central place where extension ordering is explained and link to it before we leave that here only quarter-baked ?

* composer.json: require or suggests section (composer mode)
* ext_emconf.php: depends or depends section (classic mode)

Reviewer: We could link to https://docs.typo3.org/permalink/t3coreapi:confval-ext-emconf-constraints (classic). We also write about it in https://docs.typo3.org/permalink/t3coreapi:ext-composer-json-minimal but maybe not as extended as we should.

Author: Good point. I suggest we should have a dedicated section under https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ExtensionArchitecture/BestPractises/Index.html about extension load order? Is this a way to go? If yes, I can take care of writing the docs for that section.

Reviewer: That would be great! and then I would link it from ext-emconf and composer json links.

Reviewer: @derhansen please open an issue or PR about this

> This can be achieved by requiring the original extension as a dependency in the :php:`composer.json` file

Reviewer: ```suggestion
This can be achieved by requiring the original extension as a dependency in the :file:`composer.json` file
```
